pkg/client/cache/imagestream.go

```diff
@@ -0,0 +1,101 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/labels"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// StoreToImageStreamLister gives a store List and Exists methods. The store must contain only image streams.
+type StoreToImageStreamLister struct {
+	cache.Indexer
+}
+
+// Exists checks if the given image stream exists in the store.
+func (s *StoreToImageStreamLister) Exists(stream *imageapi.ImageStream) (bool, error) {
+	_, exists, err := s.Indexer.Get(stream)
+	return exists, err
+}
+
+// List all image streams in the store.
+func (s *StoreToImageStreamLister) List() ([]imageapi.ImageStream, error) {
+	streams := []imageapi.ImageStream{}
+	for _, obj := range s.Indexer.List() {
+		streams = append(streams, *(obj.(*imageapi.ImageStream)))
+	}
+	return streams, nil
+}
+
+// GetStreamsForDeploymentConfig returns all the image streams for the provided deployment config.
+func (s *StoreToImageStreamLister) GetStreamsForDeploymentConfig(dc *deployapi.DeploymentConfig) ([]*imageapi.ImageStream, error) {
+	streams := []*imageapi.ImageStream{}
+
+	for _, key := range deployutil.ImageStreamKeysFor(dc) {
```

Comment:
The presence of this method in client I think is a code smell - this function needs to be someplace else.  client should not bring in deployutil.